presto-iceberg/src/main/java/com/facebook/presto/iceberg/procedure/RemoveOrphanFiles.java

```diff
@@ -0,0 +1,226 @@
+                if (!processedManifestFilePaths.add(manifest.path())) {
+                    // Already read this manifest
+                    continue;
+                }
+
+                validMetadataFileNames.add(extractFileName(manifest.path()));
+                try (ManifestReader<? extends ContentFile<?>> manifestReader = readerForManifest(icebergTable, manifest)) {
+                    for (ContentFile<?> contentFile : manifestReader) {
+                        validDataFileNames.add(extractFileName(contentFile.path().toString()));
+                    }
+                }
+                catch (IOException e) {
+                    throw new PrestoException(ICEBERG_FILESYSTEM_ERROR, "Unable to list manifest file content from " + manifest.path(), e);
+                }
+            }
+        }
+
+        metadataFileLocations(icebergTable, false).stream()
+                .map(this::extractFileName)
+                .forEach(validMetadataFileNames::add);
+
+        statisticsFilesLocations(icebergTable).stream()
+                .map(this::extractFileName)
+                .forEach(validMetadataFileNames::add);
+
+        validMetadataFileNames.add("version-hint.text");
+
+        // Remove unused metadata and data files older than 3 days by default
+        long expiration = olderThan == null ?
+                System.currentTimeMillis() - DAYS.toMillis(REMOVE_UNUSED_FILES_OLDER_THAN_IN_DAYS) :
+                olderThan.isLegacyTimestamp() ? olderThan.getMillisUtc() : olderThan.getMillis();
+        scanAndDeleteInvalidFiles(icebergTable, clientSession, schemaTableName, expiration, validDataFileNames.build(), "data");
+        scanAndDeleteInvalidFiles(icebergTable, clientSession, schemaTableName, expiration, validMetadataFileNames.build(), "metadata");
+    }
+
+    private void scanAndDeleteInvalidFiles(Table table, ConnectorSession session, SchemaTableName schemaTableName, long expiration, Set<String> validFiles, String subFolder)
+    {
+        try {
+            List<String> filesToDelete = new ArrayList<>();
+            FileSystem fileSystem = getFileSystem(session, this.hdfsEnvironment, schemaTableName, new Path(table.location()));
+            Path subFolderPath = new Path(table.location(), subFolder);
+            if (!fileSystem.exists(subFolderPath)) {
+                return;
+            }
+            RemoteIterator<LocatedFileStatus> allFiles = fileSystem.listFiles(subFolderPath, true);
+            while (allFiles.hasNext()) {
+                LocatedFileStatus entry = allFiles.next();
+                if (entry.getModificationTime() <= expiration && !validFiles.contains(entry.getPath().getName())) {
+                    filesToDelete.add(entry.getPath().toString());
+                    if (filesToDelete.size() >= BATCH_DELETE_FILES_COUNT) {
+                        LOGGER.debug("Deleting %s files while removing orphan files for table %s", filesToDelete, schemaTableName);
```

Comment:
I wonder if it might be better to move the file listing to the end of the log message for ease of reading since the file names can be quite long. Something like

```suggestion
                        LOGGER.debug("Deleting the following orphan files for table %s: %s", filesToDelete, schemaTableName);
```

Reply:
Thanks for the review and suggestion, fixed! Please take a look when convenient.